rpctest/rpc_harness_test.go

```diff
@@ -0,0 +1,454 @@
+
+	// The internal wallet of this harness should now have 250 BTC.
+	expectedBalance := btcutil.Amount(250 * btcutil.SatoshiPerBitcoin)
+	walletBalance := harness.ConfirmedBalance()
+	if expectedBalance != walletBalance {
+		t.Fatalf("wallet balance incorrect: expected %v, got %v",
+			expectedBalance, walletBalance)
+	}
+
+	// Now connect this local harness to the main harness then wait for
+	// their chains to synchronize.
+	if err := ConnectNode(r, harness); err != nil {
+		t.Fatalf("unable to connect harnesses: %v", err)
+	}
+	nodeSlice := []*Harness{r, harness}
+	if err := JoinNodes(nodeSlice, Blocks); err != nil {
+		t.Fatalf("unable to join node on block height: %v", err)
+	}
+
+	// The original wallet should now have a balance of 0 BTC as its entire
+	// chain should have been decimated in favor of the main harness'
+	// chain.
+	expectedBalance = btcutil.Amount(0)
+	walletBalance = harness.ConfirmedBalance()
+	if expectedBalance != walletBalance {
+		t.Fatalf("wallet balance incorrect: expected %v, got %v",
+			expectedBalance, walletBalance)
+	}
+}
+
+var harnessTestCases = []HarnessTestCase{
+	testSendOutputs,
+	testConnectNode,
+	testActiveHarnesses,
+	testJoinMempools,
+	testJoinBlocks,
+	testGenerateAndSubmitBlock,
+	testMemWalletReorg,
+}
+
+var mainHarness *Harness
+
+const (
+	numMatureOutputs = 25
+)
+
+func TestMain(m *testing.M) {
+	var err error
+	mainHarness, err = New(&chaincfg.SimNetParams, nil, nil)
+	if err != nil {
+		fmt.Println("unable to create main harness: ", err)
+		os.Exit(1)
+	}
+
+	// Initialize the main mining node with a chain of length 125,
+	// providing 25 mature coinbases to allow spending from for testing
+	// purposes.
+	if err = mainHarness.SetUp(true, numMatureOutputs); err != nil {
+		fmt.Println("unable to setup test chain: ", err)
+		os.Exit(1)
+	}
+
+	exitCode := m.Run()
+
+	// Clean up any active harnesses that are still currently running.
+	if len(ActiveHarnesses()) > 0 {
+		if err := TearDownAll(); err != nil {
+			fmt.Println("unable to tear down chain: ", err)
+			os.Exit(1)
+		}
+	}
+
+	os.Exit(exitCode)
+}
+
+func TestHarness(t *testing.T) {
+	// We should have 1250 BTC of mature unspendable outputs.
+	expectedBalance := btcutil.Amount(1250 * btcutil.SatoshiPerBitcoin)
+	harnessBalance := mainHarness.ConfirmedBalance()
+	if harnessBalance != expectedBalance {
+		t.Fatalf("expected wallet balance of %v instead have %v",
+			expectedBalance, harnessBalance)
+	}
+
+	// Current tip should be at a height of numMatureOutputs plus the
+	// required number of blocks for coinbase maturity.
+	nodeInfo, err := mainHarness.Node.GetInfo()
+	if err != nil {
+		t.Fatalf("unable to execute getinfo on node: %v", err)
+	}
+	expectedChainHeight := numMatureOutputs + uint32(mainHarness.ActiveNet.CoinbaseMaturity)
+	if uint32(nodeInfo.Blocks) != expectedChainHeight {
+		t.Errorf("Chain height is %v, should be %v",
+			nodeInfo.Blocks, expectedChainHeight)
+	}
+
+	for _, testCase := range harnessTestCases {
```

Comment:
Fair enough.  It's either here or the slice that grows linearly each time a new test cases is added.  Pretty much the same to me.  Since you prefer the slice, let's leave it.